### PR TITLE
As requested, this is a PR for 4005 vs2019build, which fixes a few things on Windows.

### PR DESCRIPTION
**Pull request by jrobcary**
Luc wrote

@jrobcary we are not testing against windows compilers in our testing.
Can you propose a PR for us to look at while we are looking at how we will test this.
If you can build a release version with ETI, tests and examples on that would be a good place to start.

so this is the response.  

I am unclear on how to build kokkos-kernels, as I only build all of trilinos.  Are there instructions on how to build kokkos-kernels?

### CODE REVIEW

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           WARNING: NO REVIEWERS HAVE BEEN REQUESTED FOR THIS PULL REQUEST!

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           WARNING: NO REVIEWERS HAVE BEEN REQUESTED FOR THIS PULL REQUEST!

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           WARNING: NO REVIEWERS HAVE BEEN REQUESTED FOR THIS PULL REQUEST!

File: src/common/KokkosKernels_BitUtils.hpp [hunk 1]
```diff
@@ -46,6 +46,10 @@
 #define _KOKKOSKERNELS_BITUTILS_HPP
 #include "Kokkos_Core.hpp"
 
+#if defined (_MSC_VER)
```
Comment (ndellingwood) [inline src/common/KokkosKernels_BitUtils.hpp:49]: I'm wondering if `_MSC_VER` should be changed to the Kokkos macro `KOKKOS_COMPILER_MSVC` from here:
https://github.com/kokkos/kokkos/blob/98b8daf4a5644c3166123ded7bfa5b4d0264399b/core/src/Kokkos_Macros.hpp#L190 

@lucbv @brian-kelley do you have a preference on macro here?

File: src/common/KokkosKernels_BitUtils.hpp [hunk 2]
```diff
@@ -203,6 +207,36 @@ int pop_count( long long i ){
   return __popcnt8(i);
 }
 
+#elif defined (_MSC_VER)
```
Comment (ndellingwood) [inline src/common/KokkosKernels_BitUtils.hpp:210]: Like above, I'm wondering if `_MSC_VER` should be changed to the Kokkos macro `KOKKOS_COMPILER_MSVC` from here:
https://github.com/kokkos/kokkos/blob/98b8daf4a5644c3166123ded7bfa5b4d0264399b/core/src/Kokkos_Macros.hpp#L190

File: src/common/KokkosKernels_BitUtils.hpp [hunk 3]
```diff
@@ -328,6 +362,35 @@ int least_set_bit(  long long i ){
   return __builtin_ffsll(i);
 }
 
+#elif defined (_MSC_VER)
```
Comment (ndellingwood) [inline src/common/KokkosKernels_BitUtils.hpp:365]: Like above, I'm wondering if `_MSC_VER` should be changed to the Kokkos macro `KOKKOS_COMPILER_MSVC` from here:
https://github.com/kokkos/kokkos/blob/98b8daf4a5644c3166123ded7bfa5b4d0264399b/core/src/Kokkos_Macros.hpp#L190

**Review by ndellingwood — COMMENTED** [review]
@lucbv should be cherry-picked to release-candidate-3.4.1 for related work within Trilinos? We won't be able to test the Windows-related changes locally (no testing system with the MSVC env that I'm aware of), but the if-guards will shield regularly tested code

**Comment by ndellingwood** [comment]
Cross-referencing #957

**Comment by jrobcary** [comment]
Would you like me to make this change?

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS BEEN REVIEWED, BUT NOT ACCEPTED OR REQUIRES CHANGES

File: src/common/KokkosKernels_BitUtils.hpp (hunk 1 above)
Comment (lucbv) [reply src/common/KokkosKernels_BitUtils.hpp:49]: Let us use the Kokkos version for consistency

File: src/common/KokkosKernels_BitUtils.hpp (hunk 2 above)
Comment (lucbv) [reply src/common/KokkosKernels_BitUtils.hpp:210]: Let's change to `KOKKOS_COMPILER_MSVC`

File: src/common/KokkosKernels_BitUtils.hpp (hunk 3 above)
Comment (lucbv) [reply src/common/KokkosKernels_BitUtils.hpp:365]: same

**Comment by lucbv** [comment]
@jrobcary yes if you could make the changes that would be nice, there is a chance that Kokkos might make changes in the future for interoperability so it would be better to benefit from that by using the variable they define.

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS NOT BEEN REVIEWED YET!

**Comment by ndellingwood** [comment]
@jrobcary could you clean up the commit history by rebasing your updates SHA db992ca6d4d0620974bd7d372ab0f4aa58974e6d to the top of the develop branch? This should reduce the commit history down to just that one commit with your changes and eliminate the other 12 commits in the commit history which will resolve the conflicting files for merge

**Comment by jrobcary** [comment]
I think this is cleaned up now.

**Comment by jrobcary** [comment]
Modified PR is #981, so closing this one.

**Pull request closed without merging**